### PR TITLE
Nodes page

### PR DESCRIPTION
**Pull request by edmundoa**
Changes to make nodes list page work again without play. This is still a work in progress, so please do not merge yet.

Currently the list of nodes is loaded (but not updated). Each node displays its metrics, and some actions are already working.

### CODE REVIEW

**Comment by garybot2** [comment]
Looks like we would not be able to merge this PR because of merge conflicts. Please rebase, fix conflicts, and force push to your branch.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterLoadBalancerStatusResource.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    private final String authenticationToken;
+
+    @Inject
+    public ClusterLoadBalancerStatusResource(NodeService nodeService,
+                                             RemoteInterfaceProvider remoteInterfaceProvider,
+                                             @Context HttpHeaders httpHeaders) throws NodeNotFoundException {
+        this.nodeService = nodeService;
+        this.remoteInterfaceProvider = remoteInterfaceProvider;
+
+        final List<String> authenticationTokens = httpHeaders.getRequestHeader("Authorization");
```
Comment (dennisoelkers) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterLoadBalancerStatusResource.java]: I think we should extract the authentication token extraction into the constructor of a base class `ProxiedResource` that is subclassed by all resources similar to this one and just make `authenticationToken` protected and use that one in the subclasses.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterSystemProcessingResource.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    private final String authenticationToken;
+
+    @Inject
+    public ClusterSystemProcessingResource(NodeService nodeService,
+                                           RemoteInterfaceProvider remoteInterfaceProvider,
+                                           @Context HttpHeaders httpHeaders) throws NodeNotFoundException {
+        this.nodeService = nodeService;
+        this.remoteInterfaceProvider = remoteInterfaceProvider;
+
+        final List<String> authenticationTokens = httpHeaders.getRequestHeader("Authorization");
```
Comment (dennisoelkers) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterSystemProcessingResource.java]: See comment above.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterSystemResource.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    private final String authenticationToken;
+
+    @Inject
+    public ClusterSystemResource(NodeService nodeService,
+                                 RemoteInterfaceProvider remoteInterfaceProvider,
+                                 @Context HttpHeaders httpHeaders) throws NodeNotFoundException {
+        this.nodeService = nodeService;
+        this.remoteInterfaceProvider = remoteInterfaceProvider;
+
+        final List<String> authenticationTokens = httpHeaders.getRequestHeader("Authorization");
```
Comment (dennisoelkers) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterSystemResource.java]: See comment above.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterSystemShutdownResource.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    private final String authenticationToken;
+
+    @Inject
+    public ClusterSystemShutdownResource(NodeService nodeService,
+                                         RemoteInterfaceProvider remoteInterfaceProvider,
+                                         @Context HttpHeaders httpHeaders) throws NodeNotFoundException {
+        this.nodeService = nodeService;
+        this.remoteInterfaceProvider = remoteInterfaceProvider;
+
+        final List<String> authenticationTokens = httpHeaders.getRequestHeader("Authorization");
```
Comment (dennisoelkers) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterSystemShutdownResource.java]: See comment above.

File: graylog2-web-interface/src/components/nodes/JvmHeapUsage.jsx [hunk 5]
```diff
@@ -1,102 +1,86 @@
...
+            <strong><span className="heap-total"> {numeral(metrics.committedMemory).format('0.0 b')}</span></strong>
+            {' '}heap space and will not attempt to use more than{' '}
+            <span className="blob" style={{backgroundColor: '#f5f5f5'}}/>
+            <strong><span className="heap-max"> {numeral(metrics.maxMemory).format('0.0 b')}</span></strong>
+          </p>
+        );
+      }
+    } else {
+      progressBar = <div className="progress"></div>;
+      detail = <p><i className="fa fa-spin fa-spinner"/> Loading heap usage information...</p>;
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/nodes/JvmHeapUsage.jsx]: You can reuse the `components/common/Spinner` component for this and pass a `text` prop.

File: graylog2-web-interface/src/components/nodes/JvmHeapUsage.jsx [hunk 6]
```diff
@@ -1,102 +1,86 @@
...
+
+    if (this.state.metrics) {
+      const metrics = this._extractMetricValues();
+
+      if (Object.keys(metrics).length === 0) {
+        progressBar = <div className="progress"></div>;
+        detail = <p>Heap information unavailable.</p>;
+      } else {
+        progressBar = (
+          <div className="progress">
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/nodes/JvmHeapUsage.jsx]: Can we reuse the react-bootstrap `ProgressBar` component for this?

File: graylog2-web-interface/src/components/throughput/NodeThroughput.jsx [hunk 7]
```diff
@@ -1,60 +1,70 @@
...
+      return (
+        <span>
+          In {numeral(metrics.totalIn).format('0,0')} / Out {numeral(metrics.totalOut).format('0,0')} msg/s.
+        </span>
+      );
     }
+  },
+  render() {
+    if (this._isLoading()) {
+      return (<span><i className="fa fa-spin fa-spinner"/> Loading throughput...</span>);
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/throughput/NodeThroughput.jsx]: You can reuse the components/common/Spinner component for this and pass a text prop.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterLoadBalancerStatusResource.java (hunk 1 above)
Comment (edmundoa) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterLoadBalancerStatusResource.java]: Sounds good! Will do that for the classes in the PR, and we can migrate the others once it's merged :)

File: graylog2-web-interface/src/components/nodes/JvmHeapUsage.jsx (hunk 5 above)
Comment (edmundoa) [inline graylog2-web-interface/src/components/nodes/JvmHeapUsage.jsx]: Yes, didn't think about that.

File: graylog2-web-interface/src/components/nodes/JvmHeapUsage.jsx (hunk 6 above)
Comment (edmundoa) [inline graylog2-web-interface/src/components/nodes/JvmHeapUsage.jsx]: I basically left the code as it was, will play a bit with `ProgressBar` and see if it makes it easier or better.

**Pull request merged**